homeassistant/components/notion/diagnostics.py

```diff
@@ -0,0 +1,32 @@
+"""Diagnostics support for Notion."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+CONF_DEVICE_KEY = "device_key"
+
+TO_REDACT = {
+    CONF_DEVICE_KEY,
+    CONF_EMAIL,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    return {
+        "data": async_redact_data(coordinator.data, TO_REDACT),
+    }
```

Comment:
The data key is no longer needed here as it only consists of one item in the dictionary that is returned